pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2108,8 +2108,11 @@ type ExecutorConfig struct {
 type ScriptTemplate struct {
 	apiv1.Container `json:",inline" protobuf:"bytes,1,opt,name=container"`
 
+	// Arguments to be passed to the script
+	SourceArgs []string `json:"sourceArgs" protobuf:"bytes,2,opt,name=sourceArgs"`
```

Comment:
anyway we can avoid a new item? we already have args

Reply:
If we want to preserve the old behavior and have an alternative to pass arguments to the script, I don't think we have the choice. If you don't want to add a new field, It's a matter of choosing between the old and the new behavior.

Reply:
I don't think a new field for args will pass review I'm afraid (I maybe wrong).

Basically, we need a flag to determine where the script goes in the args, start or end.

How about:

`AppendScript` *bool

* nil or true - > it is appended to the end - this would be default behaviour
* false -> it is inserted at the start

Reply:
I can do that this would fix my issue and avoid me the patch on our version of argo, but with this kind of option a user can't choose at the same time the args of the image's entrypoint and the args of the script. This is not something I am interested in but this will reduce the possibilities further.

Also It feels a bit weird to break by default the behavior of kubernetes that execute the command and then append args and adding a new field like I did would resolve this.

Either way If we settle on something I can easily do the change.

Reply:
what about supporting some kind of parametreizabel arg?

E.g 

```
args: ["exec", "${argo.script.path}"]
```

Reply:
I like this approach! I updated my PR, let me know if you find this better. I will try to add some docs for this behavior after that.